movies/routes/celebrities.js

```diff
@@ -0,0 +1,87 @@
+const express = require("express");
+const Celebrity = require("../models/celebrities");
+const router = express.Router();
+
+router.get("/celebrities", (req, res, next) => {
+  Celebrity.find({}, (err, celebrities) => {
+    if (err) return next(err);
+    res.render("celebrities/index", {
+      celebrities
+    });
+  });
+});
+
+
+router.get("/celebrities/new", (req, res, next) => {
+  res.render("celebrities/new");
+});
+
+router.post("/celebrities", (req, res, next) => {
+  const celebInfo = {
+    name: req.body.name,
+    occupation: req.body.occupation,
+    catchPhrase: req.body.catchPhrase
+  };
+  const newCeleb = new Celebrity(celebInfo);
+
+  newCeleb.save(err => {
+    if (err) {
+      return next(err);
+    }
+    return res.redirect("/celebrities");
+  });
+});
+
+
+router.get("/celebrities/:id", (req, res, next) => {
+  const celebId = req.params.id;
+  Celebrity.findById(celebId, (err, celebrity) => {
+    console.log(celebrity);
+    res.render("celebrities/show", {
+      celebrity
+    });
+  });
+});
+
+router.post("/celebrities/:id/delete", (req, res, next) => {
+  const celebId = req.params.id;
+  Celebrity.findByIdAndRemove(celebId, err => {
+    if (err) {
+      return next(err);
+    }
+    console.log("successfully removed:", celebId);
+    res.redirect("/celebrities");
+  });
+});
+
+
+
+router.post("/celebrities/:id", (req, res, next) => {
+  const celebId = req.params.id;
+  const updatedCelebInfo = {
+    name: req.body.name,
+    occupation: req.body.occupation,
+    catchPhrase: req.body.catchPhrase,
+  };
+  Celebrity.update({_id: celebId}, updatedCelebInfo, (err, celebrity) => {
```

Comment:
The reason you need to pass an object as an argument here, is that you are using the mongoose Model.update method, which "Updates all documents matching conditions using the update clause." as such:

var conditions = { name: 'bourne' }
  , update = { $inc: { visits: 1 }}
  , options = { multi: true };
Model.update(conditions, update, options, callback);

see http://mongoosejs.com/docs/2.7.x/docs/updating-documents.html

This method expects an object with a list of properties to match, while the method used in the learning unit is a similar but different Model.findByIdAndUpdate which expects an obejct id as argument. Always make sure you are passing the query what it expects :)

Reply:
That makes a lot of sense, I just didin't notice that we used 2 different methods. Thanks a lot for the clarification!